src/libcollections/vec.rs

```diff
@@ -1039,18 +1039,24 @@ impl<T> Vec<T> {
     #[inline]
     #[stable(feature = "append", since = "1.4.0")]
     pub fn append(&mut self, other: &mut Self) {
-        self.reserve(other.len());
-        let len = self.len();
-        unsafe {
-            ptr::copy_nonoverlapping(other.as_ptr(), self.get_unchecked_mut(len), other.len());
-        }
-
-        self.len += other.len();
         unsafe {
+            let count = other.len();
+            self.append_elements(other.as_ptr(), count);
             other.set_len(0);
         }
     }
 
+    /// Appends a elements to `Self`, without moving.
```

Comment:
(Please change this comment to say that it moves the elements, or at least not say without moving, because it is moving them.)